### PR TITLE
Gear routes and ad-hoc UI enablers

### PR DESCRIPTION
**Pull request by kofalt**
Add `/gears` and `/gears/x` routes to the API.

Must not be merged until #296 is solved here.
Note to @rentzso - feel free to put the change for that on this branch.

### CODE REVIEW

File: api/jobs/gears.py [hunk 1]
```diff
@@ -6,27 +6,44 @@
...
 
-    gear_doc  = config.db.singletons.find_one({'_id': 'gears'})
-    return gear_doc['gear_list']
+    projection = { }
+
+    if fields is None:
+        fields = [ ]
+        projection = { SINGLETON_KEY: 1 }
+    else:
+        fields.append('name')
```
Comment (nagem) [inline api/jobs/gears.py]: Does mongo error if you specify a projected field twice?

File: api/jobs/handlers.py [hunk 2]
```diff
@@ -7,12 +7,121 @@
...
+                        },
+                    },
+                }
+
+        """
+
+        if self.public_request:
+            self.abort(403, 'Request requires login')
+
+        return get_gear_by_name(_id)
```
Comment (nagem) [inline api/jobs/handlers.py]: Probably want to throw a 404 if the return is None.

Comment (nagem) [inline api/jobs/handlers.py]: I see you throw an exception in `get_gear_by_name` if it is not found. Maybe add a try around here so you can throw the proper status code.

**Comment by nagem** [comment]
LGTM pending comments.

File: api/jobs/gears.py (hunk 1 above)
Comment (kofalt) [inline api/jobs/gears.py]: I checked - it does not! Otherwise I could do the `list(set(x))` trick.

**Comment by kofalt** [comment]
Renamed a few redundant variables, and added tag filtering to `/jobs/next`. Closes #299.

**Comment by kofalt** [comment]
Rebased to resolve conflicts with #304

**Comment by rentzso** [comment]
@kofalt just pushed a new commit that should solve #296 
I haven't tested it though. Could you give it a try?

If it doesn't work, please provide some test scripts that I can use to reproduce.

**Comment by kofalt** [comment]
Thanks! I'll have a go at it.

File: bin/database.py [hunk 3]
```diff
@@ -257,6 +257,50 @@ def upgrade_to_9():
...
+    FR.container_id   --> id
+    FR.filename       --> name
+
+    CR.container_type --> type
+    CR.container_id   --> id
+    """
+
+    def switch_keys(doc, x, y):
+        doc[y] = doc[x]
+        doc.pop(x, None)
```
Comment (nagem) [inline bin/database.py]: you could also one line it like:

`doc[y] = doc.pop(x, None)`

Comment (kofalt) [inline bin/database.py]: Heh, you're not wrong...

**Comment by nagem** [comment]
New changes LGTM.

**Pull request merged**